src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Registration/RegistrationExistsValidator.cs

```diff
@@ -30,6 +30,15 @@ public override Task<bool> ShouldRunLeaf(ValidationContext data, PackageRegistra
 
             if (v2Exists != v3Exists)
             {
+                // Currently, leaf nodes are not deleted after a package is deleted.
+                // This is a known bug. Do not fail validations because of it.
+                // See https://github.com/NuGet/NuGetGallery/issues/4475
+                if (!v2Exists && v3Exists && !(v2 is PackageRegistrationIndexMetadata) && !(v3 is PackageRegistrationIndexMetadata))
```

Comment:
You already know v2Exists != v3Exists, so do you only need to check one of these here?

Reply:
yes, very true